presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -183,36 +182,20 @@ public ThriftMetastoreStats getStats()
     }
 
     @Override
-    public Optional<List<String>> getAllTables(String databaseName)
+    public List<String> getAllTables(String databaseName)
     {
-        Callable<List<String>> getAllTables = stats.getGetAllTables().wrap(() -> {
-            try (ThriftMetastoreClient client = clientProvider.createMetastoreClient()) {
-                return client.getAllTables(databaseName);
-            }
-        });
-
-        Callable<Void> getDatabase = stats.getGetDatabase().wrap(() -> {
-            try (ThriftMetastoreClient client = clientProvider.createMetastoreClient()) {
-                client.getDatabase(databaseName);
-                return null;
-            }
-        });
-
         try {
             return retry()
                     .stopOn(NoSuchObjectException.class)
                     .stopOnIllegalExceptions()
                     .run("getAllTables", () -> {
-                        List<String> tables = getAllTables.call();
-                        if (tables.isEmpty()) {
-                            // Check to see if the database exists
-                            getDatabase.call();
+                        try (ThriftMetastoreClient client = clientProvider.createMetastoreClient()) {
+                            return client.getAllTables(databaseName);
                         }
-                        return Optional.of(tables);
                     });
         }
         catch (NoSuchObjectException e) {
```

Comment:
it looks like this exception was supposed to be raised by `getDatabase.call();`. So we could try to remove it this catch.

Reply:
I am not sure we have sufficient test coverage for various metastore versions. E.g. even if we run product tests with CDH, the product tests will likely not cover this path. So, I am reluctant to remove this `catch`.